diff --git a/includes/DraftHooks.php b/includes/DraftHooks.php
index 3304c74..66bcd7f 100644
--- a/includes/DraftHooks.php
+++ b/includes/DraftHooks.php
@@ -7,7 +7,16 @@
  */
 
 use MediaWiki\MediaWikiServices;
+use MediaWiki\User\User;
 use MediaWiki\User\UserIdentity;
+use MediaWiki\EditPage\EditPage;
+use MediaWiki\Title\Title;
+use MediaWiki\Page\WikiPage;
+use MediaWiki\Linker\Linker;
+use MediaWiki\Html\Xml;
+use MediaWiki\Utils\MWCryptRand;
+use MediaWiki\Status\Status;
+
 
 class DraftHooks {
 	/**
