diff --git a/redisJobChronService b/redisJobChronService
index 5de5484..b533c97 100755
--- a/redisJobChronService
+++ b/redisJobChronService
@@ -9,7 +9,7 @@ require __DIR__ . '/src/RedisJobChronService.php';
 
 RedisJobService::checkEnvironment();
 
-error_reporting( E_ALL | E_STRICT );
+error_reporting( E_ALL );
 ini_set( 'display_errors', 1 );
 
 // Run the server...
diff --git a/redisJobRunnerService b/redisJobRunnerService
index 2ee8531..7530547 100755
--- a/redisJobRunnerService
+++ b/redisJobRunnerService
@@ -9,7 +9,7 @@ require __DIR__ . '/src/RedisJobRunnerService.php';
 
 RedisJobService::checkEnvironment();
 
-error_reporting( E_ALL | E_STRICT );
+error_reporting( E_ALL );
 ini_set( 'display_errors', 1 );
 
 // Run the server...
