diff --git a/includes/search/RevisionSearchResultTrait.php b/includes/search/RevisionSearchResultTrait.php
index fe8ec226..36b96a66 100644
--- a/includes/search/RevisionSearchResultTrait.php
+++ b/includes/search/RevisionSearchResultTrait.php
@@ -44,7 +44,7 @@ trait RevisionSearchResultTrait {
 	 */
 	protected function initFromTitle( $title ) {
 		$this->mTitle = $title;
-		if ( $title !== null ) {
+		if ( $title !== null && $title->canExist() ) {
 			$services = MediaWikiServices::getInstance();
 			$id = false;
 			( new HookRunner( $services->getHookContainer() ) )->onSearchResultInitFromTitle( $title, $id );
